### PR TITLE
feat: #101 Developers should be directed to information if trying to access the marketplace without a client Id

### PR DESCRIPTION
**Pull request by nphivu414**
This PR is related to 2 tickets [#101](https://github.com/reapit/foundations/issues/101) and [#99](https://github.com/reapit/foundations/issues/99)

We will change the behavior when a user login to client portal with a developer account or login to developer portal with a client account. We don't throw or log the error in Sentry anymore. Instead, show a modal to redirect users to the respective routes.

### CODE REVIEW

File: packages/marketplace/src/core/private-route.tsx [hunk 1]
```diff
@@ -1,36 +1,76 @@
...
+    return false
+  }
+  const { clientId, developerId } = loginIdentity
+  if (!clientId && !developerId) {
+    return true
+  }
+  return false
+}
+
+export const handleChangeLoginType = (loginIdentity, loginType, allow, dispatch) => {
```
Comment (willmcvay) [inline packages/marketplace/src/core/private-route.tsx:29]: Can you add definitions please, and to the handleRedirect function

**Comment by willmcvay** [comment]
Looks good, just the small things above but the logic looks sound and clean solution 👍

File: packages/marketplace/src/components/pages/authentication/__test__/authentication.test.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  renderClientModal,
+  onLogoutButtonClick,
+  onMarketplaceButtonClick,
+  onRegisterButtonClick,
+  onDevelopersButtonClick,
+} from '../authentication'
+import { authLogout } from '@/actions/auth'
+
+describe('Authentication', () => {
+  const history = {
```
Comment (tanphamhaiduong) [inline packages/marketplace/src/components/pages/authentication/__test__/authentication.test.tsx:19]: I think there is mockRouter on the project, could you find it?

File: packages/marketplace/src/components/pages/authentication/authentication.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      footerItems={
+        <>
+          <Button onClick={onRegisterButtonClick(history)}>Register</Button>
+          <Button onClick={onMarketplaceButtonClick(history)}>Marketplace</Button>
+          <Button onClick={onLogoutButtonClick(dispatch)}>Logout</Button>
+        </>
+      }
+    >
+      <p>
+        {'You are not currently registered as a Reapit Developer, if you would like to register please visit '}{' '}
```
Comment (tanphamhaiduong) [inline packages/marketplace/src/components/pages/authentication/authentication.tsx:84]: I think can remove {' '}
```
You are not currently registered as a Reapit Developer, if you would like to register please visit&nbsp;
```

File: packages/marketplace/src/constants/routes.ts [hunk 4]
```diff
@@ -1,4 +1,6 @@
 const Routes = {
+  Authentication: '/authentication',
+  AuthenticationLoginType: '/authentication/:loginType',
   CLIENT: '/client/apps',
   CLIENT_APP_DETAIL: '/client/apps/:appid',
```
Comment (vuhuucuong) [inline packages/marketplace/src/constants/routes.ts:5]: should be uppercase following convention

File: packages/marketplace/src/core/private-route.tsx [hunk 5]
```diff
@@ -1,36 +1,82 @@
...
+}
+
+export const handleRedirectToAuthenticationPage = (loginIdentity, allow) => {
+  return () => {
+    if (!loginIdentity) {
+      return
+    }
+    const { clientId, developerId } = loginIdentity
+    if ((allow === 'CLIENT' && !clientId) || (allow === 'DEVELOPER' && !developerId)) {
+      window.location.href = `${Routes.Authentication}/${allow}`
```
Comment (vuhuucuong) [inline packages/marketplace/src/core/private-route.tsx:56]: Can we use [useHistory](https://reacttraining.com/react-router/web/api/Hooks/usehistory) from react-router instead?

File: packages/marketplace/src/constants/routes.ts (hunk 4 above)
Comment (willmcvay) [reply packages/marketplace/src/constants/routes.ts:5]: I missed this, yes, should be uppercase

**Pull request merged**